packages/alchemy/src/middleware/gas-manager.ts

```diff
@@ -112,13 +112,16 @@ const withAlchemyGasAndPaymasterAndDataMiddleware = (
       await resolveProperties(struct)
     );
 
-    let feeOverride = undefined;
-    if (userOperation.maxFeePerGas && BigInt(userOperation.maxFeePerGas) > 0n) {
-      feeOverride = {
-        maxFeePerGas: userOperation.maxFeePerGas,
-        maxPriorityFeePerGas: userOperation.maxPriorityFeePerGas,
-      };
-    }
+    const overrides =
```

Comment:
Shouldn't this use applyOverrides? Right now it doesn't account for the cases I brought up in the first review

Reply:
ah right. forgot to update here. will update

Reply:
oh so throughout the pipeline, we should not apply the percentage overrides and just pass through until here right?

Reply:
So when the override is percentage, then we should NOT call the apis for estimating gas fee, etc but just pass through with the percentage correct? @moldy530

Reply:
Currently for percentage, the middleware pipeline just applies the percentage on the estimates set from the feeDataGetter and gasEstimator, not just for alchemy provider but also for base provider. So TLDR, we would want to only make override params accept percentage types only for alchemy provider while the base provider experience stays the same right?

Reply:
I think we chatted some offline about this, but I think we should apply overrides as percent or constants during the typical override location (right before paymaster and data) in the base provider. 

Having a percent override might actually make sense even in non alchemy cases if they're using default fee estimator and they just want to tweak its output with a buffer.

The alchemy provider will be fine with this logic because it only accepts overrides if the value is >0 otherwise it doesn't pass it as an override to the server (or shouldn't)

Reply:
Agree @moldy530 I think it makes sense to have UserOperationOverrides type in aa-core, then apply the overrides from the core smart account provider, not just for alchemy provider.

@moldy530 but what we can also do is that for alchemy provider, we don't apply the percentage override but instead just passthrough just like the absolute value override because the alchemy gas and paymaster request data api do accept the percentage type as is.

cc: @dancoombs